url78eec0e0b6_Orikuro_vic_tgz-pJ8-net_nexon_vindictus_itemcomparer_inport_StaticImportJ8/src/net/nexon/vindictus/itemcomparer/inport/StaticImport.java

```diff
@@ -0,0 +1,2125 @@
+
+		ItemSetBonus sonderbonus2 = new ItemSetBonus(220, 80, 108);
+		sonderbonus2.setSta(6);
+		boni.add(sonderbonus2);
+		
+		ItemSetBonus sonderbonus3 = new ItemSetBonus(275, 100,135);
+		sonderbonus3.setSta(7);
+		boni.add(sonderbonus3);
+		
+		ItemSetBonus sonderbonus4 = new ItemSetBonus(330, 120, 162);
+		sonderbonus4.setSta(10);
+		boni.add(sonderbonus4);
+
+		ItemSet iset = new ItemSet(setname, boni);
+		iset.setUsefulChars(char_melee + " " + magic);
+
+		Gloves g = new Gloves(iset, "Gloves", 90, ItemTyp.CLOTH, 1255, 144,
+				194, 72, 13);
+		Shoes s = new Shoes(iset, "Shoes", 90, ItemTyp.CLOTH, 1255, 144, 194,
+				72, 13);
+		Helm h = new Helm(iset, "Helm", 90, ItemTyp.CLOTH, 1255, 144, 194, 72,
+				13);
+
+		Pants p = new Pants(iset, "Pants", 90, ItemTyp.CLOTH, 1355, 144, 194,
+				72, 18);
+		Armor a = new Armor(iset, "Armor", 90, ItemTyp.CLOTH, 1422, 144, 194,
+				72, 16);
+
+		shoes.add(s);
+		gloves.add(g);
+		pants.add(p);
+		armors.add(a);
+		helms.add(h);
+	}
+	
+	public static void staticImAndExport() {
+		// 60 sets
+		generateWonderland();
+		generateLightwind();
+		generateRaiders();
+		generateBlackHammer();
+		generateShiningWill();
+		generateSwift();
+		generateSilentQue();
+		generateHolyWing();
+
+		// 70 sets
+		generateChampion();
+		generateChampionSoft();
+		generateImperialHard();
+		generateImperialSoft();
+		generateRoyalGHeavy();
+		generateMhara();
+		generateEverlastingEvie();
+		generateApocryphal();
+		generateEverlastingFio();
+		generateNighthawk();
+		generatePhantomHeavy();
+		generateRoyalGuard();
+		generateNighthawkLight();
+		generatePhantomLight();
+
+		// 80 sets
+		generateSkeleton();
+		generateSkeletonWitch();
+		generatePearlClassic();
+		generateBlackPearl();
+		generateHeremon();
+		generateNoblesse();
+		generateHeremonSoft();
+		generatePirateLeather();
+		generatePirate();
+		generateNobleseWitch();
+
+		// neue sets
+		generateWEdge();
+		generateK9();
+		generateHMagnum();
+
+		// Episode 4 Season 2
+		generateMajest();
+		generateMajestLeather();
+		generateMajestSoft();
+		generateArmageddon();
+		generateArmageddonWitch();
+
+		// Lynn
+		generateSoulloop();
+		generateOriental();
+		generateDancing();
+
+		// arisha
+		generateApex();
+		generateSilverFox();
+		generateStriker();
+
+		// 90 gear
+		generateRegina_3_min();
+		generateBarha_3_min();
+		generateGuardian_3_min();
+		generateCloth_3_min();
+	}
+
+	public static void staticExportSets() {
+		HashSet<ItemSet> setnames = new HashSet<>();
+
+		for (Shoes s : shoes) {
+			setnames.add(s.getItemset());
+		}
+		for (Pants s : pants) {
+			setnames.add(s.getItemset());
+		}
+		for (Gloves s : gloves) {
+			setnames.add(s.getItemset());
+		}
+		for (Armor s : armors) {
+			setnames.add(s.getItemset());
+		}
+		for (Helm s : helms) {
+			setnames.add(s.getItemset());
+		}
+		isets = setnames;
+		try {
+			File outd = new File("data" + File.separator);
+			outd.mkdirs();
+			BufferedWriter out = new BufferedWriter(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *out*. The resource is referenced at lines 1955, 1956, 1962. The resource is closed at line 1963. There are other execution paths that do not contain closure statements, for example, when Marshaller.marshal throws an exception. Ensure *out* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).